### DISCUSSION TITLE
Resetting Zustand Jest Configs failing

### DISCUSSION THREAD

**Discussion by aqarain** [pmndrs/zustand#2606] [Q&A]
Hi 👋🏼,

In my small app, I am trying to set up Zustand.
- Without the `__mocks__/zustand.ts` file, the second test fails because I think the store is not being reset.
- But if I add the `__mocks__/zustand.ts` file (with the configs from [Zustand docs](https://docs.pmnd.rs/zustand/guides/testing#jest)), both the test fail.
 
Probably, I am missing something 😕

Can someone please have a look what could be missing or wrong?

Thank you 🙂

Stackblitz: https://stackblitz.com/edit/vitejs-vite-8lqgbb

**Comment by dbritto-dev** [comment]
@aqarain here you go -> https://stackblitz.com/edit/vitejs-vite-pocipg?file=src%2F__mocks__%2Fzustand.ts in your case since you are using useStore you have to options:

Export just useStore at the end

```ts
// at the end

export const useStore = zustand.useStore
```

Export everything at the beginning

```ts
export * from 'zustand'

const { create: actualCreate, createStore: actualCreateStore } =
  jest.requireActual<typeof zustand>('zustand');

// the rest of the code
```

**Reply by aqarain** [reply]
@dbritto-dev 
Thank you for quick response 😊
But I think something is wrong because if you log the `items` at the end of 2nd test, it should be `[ { id: 'two', name: 'two' } ]`
But it is `[ { id: 'one', name: 'one' } ]` (old items) 🤔

**Reply by dbritto-dev** [reply]
@aqarain as far I see everything looks good, what's wrong with?

**Reply by aqarain** [reply]
<img width="1186" alt="Screenshot 2024-06-13 at 20 58 20" src="https://github.com/pmndrs/zustand/assets/20900424/0afc0705-85bd-4c2c-bc82-d81485a69c7f">

@dbritto-dev You see the last assertion fails. It should pass.
As you see in the printed DOM, the value should be `two`  but it is getting the old value:
```
<li>
            one
            <button>
               🗑️ 
            </button>
          </li>
```

**Reply by dbritto-dev** [reply]
everything looks good to me

<img width="773" alt="image" src="https://github.com/pmndrs/zustand/assets/1697714/aaddb54b-9674-40bc-a53f-a652369fcb52">

**Reply by dbritto-dev** [reply]
https://stackblitz.com/edit/vitejs-vite-pocipg?file=src%2F__mocks__%2Fzustand.ts,src%2F__tests__%2FApp.test.tsx

**Reply by aqarain** [reply]
@dbritto-dev Please add the below assertion at the end of 2nd test and it will fail. 
```
expect(screen.getByText(itemName)).toBeInTheDocument();
```

It should pass the above assertion.

**Reply by dbritto-dev** [reply]
@aqarain seems the mocks are not working, would you mind creating one from our jest template -> https://stackblitz.com/edit/jest-zustand?file=README.md?

**Reply by aqarain** [reply]
@dbritto-dev Sorry, what do I need to create? I am not sure what is missing?

**Reply by dbritto-dev** [reply]
@aqarain just fork our jest template to create a new one for you, just to double check

**Reply by dbritto-dev** [reply]
@aqarain btw, I encourage to use vitest instead jest

**Reply by aqarain** [reply]
@dbritto-dev 
Using Jest-Zustand template: https://stackblitz.com/edit/jest-zustand-x1pgkz

Unfortunately, I cannot use vitest. We use Jest in our tech stack

**Reply by aqarain** [reply]
I fixed it 😆
https://stackblitz.com/edit/vitejs-vite-75vw34

**Reply by aqarain** [reply]
@dbritto-dev 
With this you don't need `__mocks__/zustand.ts`.
I was wondering why is `__mocks__/zustand.ts` needed and if I remove it, will it have any negative impact?

**Reply by dbritto-dev** [reply]
What I found was that we weren't mocking zustand/vanilla just zustand so changing that to just zustand works

**Reply by aqarain** [reply]
@dbritto-dev 
Is there a way to mock `zustand/vanilla`?
Changing to `zustand` in store does work but I have to use the store outside of React so have to go with `zustand/vanilla`

**Reply by dbritto-dev** [reply]
@aqarain yes, instead of zustand.ts you can create two files under zustand folder: index.ts and vanilla.ts

**Reply by aqarain** [reply]
I create a `zustand` folder with 2 files: `index.ts` and `vanilla.ts` but still it fails.
Can you please have a quick look?

 https://stackblitz.com/edit/vitejs-vite-8lqgbb?file=src%2Fzustand%2Findex.ts,src%2Fzustand%2Fvanilla.ts

**Reply by dbritto-dev** [reply]
@aqarain I'll review it later but you can read more about that in the jest mocking modules section
